src/anderson_darling.jl

```diff
@@ -89,99 +113,191 @@ the samples come from different distributions.
+
+"""Asymptotic evaluation of the p-value for Anderson-Darling test"""
+function pvalueasym(x::KSampleADTest)
+    # Computational Details:
+    #
+    # This function uses the upper Tₘ quantiles as obtained via simulation of
+    # the Anderson-Darling test statistics (2*10^6) with sample from standard
+    # Normal distribution with size n=500 for each sample.
+    #
+    # After standardization of AD statistics, p-quantiles were estimate
+    # for p ∈ [.00001,.00005,.0001,.0005,.001,.005,.01,.025,.05,.075,
+    # .1,.2,.3,.4,.5,.6,.7,.8,.9,.925,.95,.975,.99,.9925,.995,.9975,.999,
+    # .99925,.9995,.99975,.9999,.999925,.99995,.999975,.99999]
+    #
+    # Particular p-quantiles are determined from the simulations for various
+    # values of `m`, m ∈ [1,2,3,4,6,8,10,20]. By interpolation, for the given
+    # value of `m` from the performed test is done by fitting a cubic polynomial
+    # of 1/sqrt(ms) to the simulated Tₘ quantiles.
+    #
+    # Next, the quadratic polynomial is used to fit the log((1-p)/p) to above
+    # interpolated quantiles and the value fitted to the estimated standardized
+    # AD statistics Tₘ.
+    #
+    # The p-values from Table 1 of the original paper were reproduced with
+    # relative error bounded bounded by 1% in 85% of cases (see the relative
+    # error table below).
+    #
+    # m\α   .25      .10     .05     .025     .01
+    #  1   0.0001   0.0018  0.0006  0.0112   0.0305
+    #  2  -0.0089  -0.0064  0.0024  0.0121   0.0306
+    #  3  -0.0068  -0.0026  0.0045  0.0082   0.0164
+    #  4  -0.0042  -0.0009  0.0027  0.005    0.0079
+    #  6  -0.0025  -0.0005  0.005   0.003   -0.0004
+    #  8  -0.0024   0.0011  0.0027  0.0023  -0.0023
+    #  10 -0.0038   0.0008  0.0023  0.0026  -0.0043
+    #  ∞  -0.0038   0.0014  0.005   0.0037   0.0148
+    #
+
     m = x.k - 1
     Tk = (x.A²k - m) / x.σ
-    sig = [0.25, 0.1, 0.05, 0.025, 0.01]
-    b0 = [0.675, 1.281, 1.645, 1.96, 2.326]
-    b1 = [-0.245, 0.25, 0.678, 1.149, 1.822]
-    b2 = [-0.105, -0.305, -0.362, -0.391, -0.396]
-    tm = b0 + b1 / sqrt(m) + b2 / m
-
-    # Fit a quadratic polynomial to tm and log(sig)
-    # Adapted from code by Paulo José Salz Jabardo
-    A = zeros(eltype(tm), 5, 3)
-    A[:,1] = 1.0
-    @inbounds for i = 1:2, k = 1:5
-        A[k,i+1] = A[k,i] * tm[k]
+    sqm = 1.0 ./ sqrt.(M)
+
+    T = x.modified ? TKM : TK
+
+    # Construct Tm curve
+    n = length(PV)
+    Tm = zeros(n)
+    for i in 1:n
+        A = [ sqm[i]^p for i = 1:length(sqm), p = 0:3 ] # fit in cubic
+        C = A \ T[:,i]
+        x = 1/sqrt(m)
+        Tm[i] = C[1] + C[2]*x + C[3]*x^2 + C[4]*x^3
     end
-    f = A \ log.(sig)
-
-    exp(f[1] + f[2]*Tk + f[3]*Tk^2)
+    logP =  log.((1 .- PV) ./ PV )
+
+    # locate curve area for extrapolation
+    _, j = findmin(abs.(Tm .- Tk))
```

Comment:
Could you be more specific?

Reply:
This will make two passes over the array: one for broadcasting and one for finding the minimum. Generators are lazy, so iteration only happens on-demand, in this case when finding the minimum. This would then be
```julia
findmin(abs(tm - tk) for (tm, tk) in zip(Tm, Tk))
```
or something along those lines.